### ISSUE TITLE
bug: service id does not support string

### ISSUE THREAD

**Issue by jackwoods** [apache/incubator-apisix#1954]
### Issue description
service id does not support string

### Environment

* apisix version (cmd: `apisix version`):
* OS:

### Minimal test code / Steps to reproduce the issue
Client error: `PUT http://10.60.53.79:9080/apisix/admin/services/service-xxx-xxx` resulted in a `400 Bad Request` response:
{"error_msg":"wrong type of service id"}

### Source code
services.lua
```lua
    if need_id and not tonumber(id) then
        return nil, {error_msg = "wrong type of service id"}
    end
```

**Comment by moonming** [comment]
welcome PR to fix it.

**Comment by jackwoods** [comment]
I saw @membphis has fixed it in branch v1.4.1
